components/kyma-operator/pkg/steps/installation.go

```diff
@@ -114,15 +123,28 @@ func (steps *InstallationSteps) processComponents(installationData *config.Insta
 		step := stepsFactory.NewStep(component)
 
 		steps.PrintStep(stepName)
-
-		processErr := step.Run()
-		if steps.errorHandlers.CheckError("Step error: ", processErr) {
-			_ = steps.statusManager.Error(component.GetReleaseName(), stepName, processErr)
-			return processErr
+		backoff.reset()
+
+		firstRun := true
+		var processErr error
+
+		for processErr != nil || firstRun {
```

Comment:
How about using the [avast/retry-go](https://github.com/avast/retry-go)? I'm not sure why we didn't adopt it in the first place. Anyways, the library seems to suit our needs and we could significantly reduce the number of SLOC.

Reply:
Not sure if the number of SLOC is reduced but that definitely made the code more readable 👍